### PR TITLE
feat: refactor AT and Monitoring config

### PR DESCRIPTION
**Pull request by ocofaigh**
### Description

In 1.67.0, you no longer need to pass monitoring or AT CRNs to enable monitoring or activity tracking on a bucket. If values not passed, it will default to the Monitoring or AT instance in the same location as the bucket.

As part of this work `sysdig_crn` has been renamed to  `monitoring_crn`.

The following new variables have also been added to the root level module: `activity_tracker_read_data_events`, `activity_tracker_write_data_events`, `activity_tracker_management_events`, `request_metrics_enabled` and `usage_metrics_enabled`. Previously the logic only set these to true if an instance CRN was passed, but as that is now optional, I have exposed them all as their own variables and defaulted them all to `true`

See https://github.com/terraform-ibm-modules/terraform-ibm-cos/issues/643 and https://github.com/terraform-ibm-modules/terraform-ibm-cos/issues/648

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by ocofaigh** [comment]
Upgrade test failing:

```
        	Messages:   	Resource(s) identified to be updated 
        	            	Name: cloud_monitoring
        	            	Address: module.observability_instances.module.cloud_monitoring.ibm_resource_instance.cloud_monitoring[0]
        	            	Actions: [update]
        	            	DIFF:
        	            	  Before: 
        	            		{"name":"cos-upgrade-lsl6er-sysdig"}
        	            	  After: 
        	            		{"name":"cos-upgrade-lsl6er-monitoring"}
```
^^ expected due to name update in example - no impact to consumers of the module

```
        	Messages:   	Resource(s) identified to be updated 
        	            	Name: cos_bucket
        	            	Address: module.cos_bucket1.ibm_cos_bucket.cos_bucket[0]
        	            	Actions: [update]
        	            	DIFF:
        	            	  Before: 
        	            		{"activity_tracking":"SECURE_VALUE_HIDDEN_HASH:-dbe1e6bf792df99e92dc0fd04cab5090f7b2754c78ef4cc0fea9ff45"}
        	            	  After: 
        	            		{"activity_tracking":"SECURE_VALUE_HIDDEN_HASH:-876c24c4716726d21b36bcacaa71287499ca74a4f906e1cc181a6fb4"}
```
^ expected to the monitoring an activity tracking being enabled by default - update in place - no downtime for users

```
        	Messages:   	Resource(s) identified to be updated 
        	            	Name: cos_bucket
        	            	Address: module.cos_bucket2.ibm_cos_bucket.cos_bucket[0]
        	            	Actions: [update]
        	            	DIFF:
        	            	  Before: 
        	            		{"activity_tracking":"SECURE_VALUE_HIDDEN_HASH:-42721ab5dab883e9bf917e0cd164cb3df7480d608561d60d382921e2"}
        	            	  After: 
        	            		{"activity_tracking":"SECURE_VALUE_HIDDEN_HASH:-5c7b504eadc75cc9db4ee229aff65ee5eeaeb6ae60e0b4449f2b0969"}
```
^ expected to the monitoring an activity tracking being enabled by default - update in place - no downtime for users

```
        	Messages:   	Resource(s) identified to be updated 
        	            	Name: cos_bucket1
        	            	Address: module.cos_bucket3.ibm_cos_bucket.cos_bucket1[0]
        	            	Actions: [update]
        	            	DIFF:
        	            	  Before: 
        	            		{"activity_tracking":"SECURE_VALUE_HIDDEN_HASH:-2e81669d74cc14b778d0cea4d00a56fe8ee265e67b191beb4c37555d"}
        	            	  After: 
        	            		{"activity_tracking":"SECURE_VALUE_HIDDEN_HASH:-70393208568f9dac3435db2c35f24ba8649775e42a4ca8d9e0770772"}
```
^ expected to the monitoring an activity tracking being enabled by default - update in place - no downtime for users

File: modules/buckets/version.tf [hunk 1]
```diff
@@ -5,8 +5,9 @@ terraform {
   required_providers {
     # tflint-ignore: terraform_unused_required_providers
     ibm = {
-      source  = "ibm-cloud/ibm"
-      version = ">= 1.66.0, < 2.0.0"
+      source = "ibm-cloud/ibm"
+      # version = ">= 1.67.0-beta1, < 2.0.0"
+      version = ">= 1.66.0"
```
Comment (shemau) [inline modules/buckets/version.tf:10]: Validate this version. Remove comment about beta.

**Review by shemau — APPROVED** [review]
LGTM

**Pull request merged**